controller/linodemachine_controller.go

```diff
@@ -278,14 +278,36 @@ func (r *LinodeMachineReconciler) reconcileCreate(
 	return ctrl.Result{}, nil
 }
 
+// createInstance provisions linode instance after checking if the request will be within the rate-limits
+// Note: it takes a lock before checking for the rate limits and releases it after making request to linode API or when returning from function
+func (r *LinodeMachineReconciler) createInstance(ctx context.Context, logger logr.Logger, machineScope *scope.MachineScope, createOpts *linodego.InstanceCreateOptions) (*linodego.Instance, error, bool) {
+	ctr := util.GetPostReqCounter(machineScope.TokenHash)
+	ctr.Mu.Lock()
+	defer ctr.Mu.Unlock()
+
+	if ctr.IsPOSTLimitReached() {
+		logger.Info(fmt.Sprintf("Cannot make more requests as max requests have been made. Waiting and retrying after %v seconds", reconciler.SecondaryLinodeTooManyPOSTRequestsErrorRetryDelay))
```

Comment:
are the limits an internal thing only? if not, can you add the max request to the log?

Reply:
updated